test/unit/Types/ScalarTypesResolverTest.php

```diff
@@ -0,0 +1,324 @@
+                'expectedValue' => true,
+            ],
+        ];
+    }
+
+    /**
+     * @param mixed $value
+     * @param mixed $expectedValue
+     *
+     * @dataProvider convertSerializeDataProvider
+     */
+    public function testConvertSerializesValue(bool $deserialize, int $id, $value, $expectedValue): void
+    {
+        $convertedValue = $this->scalarTypeResolver->convert($deserialize, $id, $value);
+        Assert::assertSame($expectedValue, $convertedValue);
+    }
+
+    public function testGetTypeWithoutPatternReturnsNull(): void
+    {
+        $pattern = $this->scalarTypeResolver->getPattern(0);
+        Assert::assertNull($pattern);
+    }
+
+    public function testGetPatternReturnsPattern(): void
+    {
+        $pattern         = $this->scalarTypeResolver->getPattern(1);
+        $expectedPattern = '\d{4}-\d{2}-\d{2}';
+        Assert::assertSame($expectedPattern, $pattern);
+    }
+
+    public function testGetPhpTypeReturnsPhpType(): void
+    {
+        $phpType         = $this->scalarTypeResolver->getPhpType(0);
+        $expectedPhpType = 'string';
+        Assert::assertSame($expectedPhpType, $phpType);
+    }
+
+    public function testIsDateTimeReturnsFalseForNonDatetimeType(): void
+    {
+        $isDateTime = $this->scalarTypeResolver->isDateTime(0);
+        Assert::assertFalse($isDateTime);
+    }
+
+    public function testIsDateTimeReturnsTrueForDatetimeType(): void
+    {
+        $isDateTime = $this->scalarTypeResolver->isDateTime(1);
+        Assert::assertTrue($isDateTime);
+    }
+
+    public function testFindScalarTypeTypeIsNullReturnsStringTypeId(): void
+    {
+        $scalarType         = $this->scalarTypeResolver->findScalarType(null, '');
+        $expectedScalarType = 0;
+        Assert::assertSame($expectedScalarType, $scalarType);
+    }
+
+    public function testFindScalarTypeFormatNotSetReturnsStringTypeId(): void
+    {
+        $scalarType         = $this->scalarTypeResolver->findScalarType(Type::STRING, '');
+        $expectedScalarType = 0;
+        Assert::assertSame($expectedScalarType, $scalarType);
+    }
+
+    public function testFindScalarTypeReturnsDefaultZero(): void
+    {
+        $scalarType         = $this->scalarTypeResolver->findScalarType(Type::STRING, 'test');
+        $expectedScalarType = 0;
+        Assert::assertSame($expectedScalarType, $scalarType);
+    }
+
+    public function testFindScalarTypeReturnsType(): void
+    {
+        $scalarType         = $this->scalarTypeResolver->findScalarType(Type::STRING, 'date');
+        $expectedScalarType = 1;
+        Assert::assertSame($expectedScalarType, $scalarType);
+    }
+
+    public function testFindScalarTypeFormatNotSetReturnsType(): void
+    {
+        $scalarType         = $this->scalarTypeResolver->findScalarType(Type::NUMBER, '');
+        $expectedScalarType = 5;
+        Assert::assertSame($expectedScalarType, $scalarType);
+    }
+
+    public function tearDown(): void
```

Comment:
Everything is fine, only thing left is moving this method to the top of the class, after setUp.